### PR TITLE
fix: Advert stickiness miscalculated after dismissing sign-in-gate

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

This fixes a bug where the computed sticky-region for each advert is miscalculated on pages once a sign-in gate is dismissed. This miscalculation causes ads to collide with one another (see the Before video).

This bug arises due to the following sequence of events:
- The commercial bundle running on the page will run the [article-body-adverts](https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/article-body-adverts.ts) module. This will kick off the attempts to insert inline adverts (desktop OR mobile inline adverts, inline merchandising adverts, carrot adverts). These advert slots will be inserted into the DOM.
- A browser of an article is (sometimes) presented with a Sign In gate. This bug arises when the Sign In Gate renders on the page AFTER the first run of Spacefinder is completed. The gate will hide all adverts in the article body after the first two paragraphs.
- The user then dismisses the gate. This dispatches a [custom event](https://github.com/guardian/dotcom-rendering/blob/3ac24426c459fb8456c81b5148da6c883872189e/dotcom-rendering/src/web/components/SignInGateSelector.importable.tsx#L166-L168) to the document from DCR.
- The commercial code listens for the custom event and reinvokes the insertion of all of the article-body-adverts. This next set of Spacefinder invocations will not take into account the existing containers inserted from the previous invocations of Spacefinder. This causes duplicates of each slot to be inserted into the article body. Note that the the attempts to actually fill these with GAM adverts are de-duplicated by `addSlot`, so they remain empty, but present in the DOM.
- This throws off the computation of the sticky containers, causing them to scroll for much longer than anticipated and therefore intersect with each other.

The errors of inserting duplicate slots is surfaced via this PR: #25515.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

**Before:**

https://user-images.githubusercontent.com/8000415/194527924-f2fec7ff-4e28-4caa-b482-464ae89431a8.mov

**After:**

https://user-images.githubusercontent.com/8000415/194527997-154521aa-0e3c-42eb-90c5-879ad7b97a63.mov

## What is the value of this and can you measure success?

This will remediate an issue with adverts colliding with one another after you dismiss the Sign In Gate.

### Tested

- [X] Locally
- [ ] On CODE

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.ts [hunk 1]
```diff
@@ -34,7 +37,9 @@ const hasImages = !!window.guardian.config.page.lightboxImages?.images.length;
 const hasShowcaseMainElement =
 	window.guardian.config.page.hasShowcaseMainElement;
 
-const adSlotClassSelectorSizes = {
+const adSlotcontainerClass = 'ad-slot-container';
```
Comment (zekehuntergreen) [inline static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:40]:
```suggestion
const adSlotContainerClass = 'ad-slot-container';
```
for camelCase?

**Review by arelra — APPROVED** [review]
Nice detective work! 🔎

**Review by zekehuntergreen — APPROVED** [review]
Nice work 👏  This is quite a complicated bug!

**Comment by zekehuntergreen** [comment]
Looks like we'll have to do something similar to fix the same issue in the [consentless version](https://github.com/guardian/frontend/blob/7ab26c4ac3b5fc7387842d1d613ec0b9657e07f1/static/src/javascripts/projects/commercial/modules/consentless/dynamic/article-inline.ts#L150) of this file as well.

**Review by domlander — APPROVED** [review]

**Pull request merged**